components/PromptRemoveApp.vue

```diff
@@ -0,0 +1,374 @@
+        array.push(each.type);
+
+        return array;
+      }, []));
+
+      if (types.size > 1) {
+        return this.t('generic.resource', { count: this.toRemove.length });
+      }
+
+      const schema = this.toRemove[0]?.schema;
+
+      if ( !schema ) {
+        return `resource${ this.toRemove.length === 1 ? '' : 's' }`;
+      }
+
+      return this.$store.getters['type-map/labelFor'](schema, this.toRemove.length);
+    },
+
+    selfLinks() {
+      return this.toRemove.map((resource) => {
+        return get(resource, 'links.self');
+      });
+    },
+
+    needsConfirm() {
+      const first = this.toRemove[0];
+
+      return first?.confirmRemove;
+    },
+
+    plusMore() {
+      const remaining = this.toRemove.length - this.names.length;
+
+      return this.t('promptRemove.andOthers', { count: remaining });
+    },
+
+    // if the current route ends with the ID of the resource being deleted, whatever page this is wont be valid after successful deletion: navigate away
+    doneLocation() {
+      // if deleting more than one resource, this is happening in list view and shouldn't redirect anywhere
+      if (this.toRemove.length > 1) {
+        return null;
+      }
+      const currentRoute = this.toRemove[0].currentRoute();
+      const out = {};
+      const params = { ...currentRoute.params };
+
+      if (params.id && (params.id === this.toRemove[0]?.metadata?.name || params.id === this.toRemove[0].id)) {
+        let { name = '' } = currentRoute;
+
+        name = name.slice(0, name.indexOf('-id'));
+
+        if (params.namespace) {
+          name = name.slice(0, name.indexOf('-namespace'));
+          delete params.namespace;
+        }
+        delete params.id;
+
+        out.params = params;
+        out.name = name;
+      }
+
+      return out;
+    },
+
+    currentRouter() {
+      // ...don't need a router if there's no route to go to
+      if (!this.doneLocation) {
+        return null;
+      } else {
+        return this.toRemove[0].currentRouter();
+      }
+    },
+
+    protip() {
+      return this.t('promptRemove.protip', { alternateLabel });
+    },
+
+    deleteDisabled() {
+      const confirmFailed = this.needsConfirm && this.confirmName !== this.nameToMatch;
+
+      return this.preventDelete || confirmFailed;
+    },
+
+    ...mapState('action-menu', ['showPromptRemoveApp', 'toRemove']),
+    ...mapGetters({ t: 'i18n/t' }),
+
+    resourceNames() {
+      return this.names.reduce((res, name, i) => {
+        if (i >= 5) {
+          return res;
+        }
+        res += `<b>${ name }</b>`;
+        if (i === this.names.length - 1) {
+          res += this.plusMore;
+        } else {
+          res += i === this.toRemove.length - 2 ? ' and ' : ', ';
+        }
+
+        return res;
+      }, '');
+    },
+
+    checkForCrd() {
+      if (this.allInstalled !== null) {
+        const crd = this.allInstalled.find(installed => installed.metadata.name === `${ this.names }-crd`);
```

Comment:
We should set an annotation on the app that says what other things were auto-installed for it.  Assuming a naming-scheme exists is fragile, and not always true (there are charts which auto-install things other than CRDs).

Reply:
Yeah this is something that needs more polish @jordojordo bring this up in the next sync up and we can find a path forward